packages/babel-preset/src/utils/dedupeCSSRules.ts

```diff
@@ -0,0 +1,22 @@
+import type { CSSRulesByBucket } from '@griffel/core';
+
+/**
+ * Rules that are returned by `resolveStyles()` are not deduplicated.
+ * It's critical to filter out duplicates for build-time transform to avoid duplicated rules in a bundle.
+ */
+export function dedupeCSSRules(cssRulesByBucket: CSSRulesByBucket): CSSRulesByBucket {
+  return Object.fromEntries(
+    Object.entries(cssRulesByBucket).map(([styleBucketName, cssBucketEntries]) => {
+      if (styleBucketName === 'm') {
+        return [
+          styleBucketName,
+          cssBucketEntries.filter(
+            (entryA, index, entries) => entries.findIndex(entryB => entryA[0] === entryB[0]) === index,
+          ),
```

Comment:
Opened for better ideas on making that array unique.

Reply:
this is during build time so speed might not be an issue. However, if you did want to make this faster from a compute time PoV, you could sort the array. Doing a sort operation and removing duplicates after will be cheaper than the current way you are doing it.

Also a bigger arch comment that also might not be so important but internally it should be pretty easy to maintain order in the CSS buckets if you just did ordered insertion with a binary search, it would make insertion into the bucket slightly slower, but mean less work afterwards.

This is more or less a larger nit out of intereset 🤷, no expectation to do any of it